BonnAnalysis/ComplexInterpolation.lean

```diff
@@ -94,6 +97,77 @@ def at_height (f:ℂ → ℂ) (y:ℝ) : (Icc 0 1 : Set ℝ) → ℝ  := fun x 
 
 def sup_at_height (f: ℂ → ℂ) (y: ℝ) := sSup ((at_height f y)'' univ)
 
+def abs_sup (f: ℂ → ℂ ) := sSup ((fun z ↦ Complex.abs (f z))'' { z | z.re ∈ Icc 0 1} )
+
+lemma abs_fun_nonempty (f: ℂ → ℂ) : ((fun z ↦ Complex.abs (f z))'' { z | z.re ∈ Icc 0 1}).Nonempty := by{
+  simp
+  use 0
+  simp
+}
+
+lemma abs_fun_bounded {f:ℂ → ℂ} (h2f : IsBounded (f '' { z | z.re ∈ Icc 0 1})) : BddAbove ((fun z ↦ Complex.abs (f z))'' { z | z.re ∈ Icc 0 1}) := by{
+  simp[BddAbove, upperBounds]
+  obtain ⟨R, hR⟩:= (Metric.isBounded_iff_subset_ball 0).mp h2f
+  simp only [subset_def, mem_image] at hR
+  use R
+  simp
+  intro r z hz₁ hz₂ habs
+  rw[← habs]
+  apply le_of_lt
+  specialize hR (f z) (by use z; constructor; simp[hz₁, hz₂]; rfl)
+  simp at hR
+  exact hR
+}
+
+-- For the first version of this, we need
+#check exists_seq_tendsto_sSup
+#check tendsto_subseq_of_bounded
+
+#check Complex.eqOn_of_isPreconnected_of_isMaxOn_norm
+#check Complex.norm_eqOn_closure_of_isPreconnected_of_isMaxOn -- I believe this is the one!
+
+
+#check IsPreconnected.image
+
+#check IsPreconnected.prod --but this would require seeing ℂ as ℝ^2
+
+/- Some technical lemmas to apply the maximum modulus principle -/
+lemma strip_prod : { z:ℂ  | z.re ∈ Ioo 0 1} = (Ioo 0 1 : Set ℝ) ×ℂ univ := by{
+  ext z
+  simp[Complex.mem_reProdIm]
+}
+
+lemma clstrip_prod : {z: ℂ | z.re ∈ Icc 0 1} = (Icc 0 1 : Set ℝ) ×ℂ univ := by{
+  ext z
+  simp[Complex.mem_reProdIm]
+}
+
+
+lemma isPreconnected_strip : IsPreconnected { z : ℂ | z.re ∈ Ioo 0 1} := by{
+  have : { z : ℂ | z.re ∈ Ioo 0 1} = ⇑equivRealProdCLM.toHomeomorph ⁻¹' ((Ioo 0 1 : Set ℝ) ×ˢ  (univ: Set ℝ)) := by{
+    ext z
+    simp
+  }
+  rw[this, Homeomorph.isPreconnected_preimage Complex.equivRealProdCLM.toHomeomorph]
+  exact IsPreconnected.prod isPreconnected_Ioo isPreconnected_univ
+}
+
+lemma isOpen_strip : IsOpen { z : ℂ | z.re ∈ Ioo 0 1} := by{
+  rw[strip_prod]
+  exact IsOpen.reProdIm isOpen_Ioo isOpen_univ
+}
```

Comment:
If you want, you can turn it into a proof term as follows (tested: can be committed directly): 


```suggestion
lemma isOpen_strip : IsOpen { z : ℂ | z.re ∈ Ioo 0 1} :=
  strip_prod ▸ IsOpen.reProdIm isOpen_Ioo isOpen_univ
```

Reply:
Both proof styles are perfectly fine. I personally also prefer writing `rw + exact`.

Reply:
Thanks for the comment, I did not know (or possibly forgot) you could do that